mautic/mautic
# Mark an email as opened if a link is clicked from the email

**Description**
Email clients may not load images and thus ignore an email's tracking pixel.  However if a lead clicks a link in the email, we know they actually did open it.  So this PR ensures that the email is marked as opened if a link is clicked.  Fixes #843

**Testing**
Clear symfony's cache after applying the PR.

1. Create an email with a link.
2. Ensure the email client is configured to not load images by default (or you'll have to reset the database stat in email_stats after obtaining the link from the email)
3. Send the email to a lead
4. Obtain the link and ensure the stat didn't get marked as read in the database (I just looked at email_stats directly)
5. In an anonymous browser, click the link or copy and paste to browse to it
6. View the stat in the database to ensure is now marked as opened
